astroid/transforms.py

```diff
@@ -145,7 +145,7 @@ def unregister_transform(
         """Unregister the given transform."""
         self.transforms[node_class].remove((transform, predicate))  # type: ignore[index, arg-type]
 
-    def visit(self, module: nodes.Module) -> SuccessfulInferenceResult:
+    def visit(self, module: nodes.NodeNG) -> SuccessfulInferenceResult:
```

Comment:
From poking at this I wonder if `nodes.Module | nodes.Unknown` is better (and also fix `AstroidManager.visit_transforms`, too)?

Reply:
There is brain that calls this with a node, although it might be Unknown.
I don't think there is really a reason not to accept the base node right?

Reply:
Only reason would be if it's a symptom of a mistake, then we could throw a TypeError. It sort of does seem like a symptom of a mistake, but brains do all kinds of hacky things, so I dunno. If we're going to leave the typing this way, we should update the argument name.

Reply:
I think updating the argument name is better.

This seems like something an astroid plugin might want to do: call all transforms for visited class definitions for example. Think this is a valid pattern.

Reply:
Sounds good!